packages/core/src/config/schema.ts

```diff
@@ -330,9 +330,34 @@ const FarmConfigSchema = z
   })
   .strict();
 
-export function parseUserConfig(config: unknown) {
+export function parseUserConfig(config: unknown): UserConfig {
   try {
     const parsed = FarmConfigSchema.parse(config);
+    // Auto enable decorator by default when `script.decorators` is enabled
+    if (parsed.compilation?.script?.decorators !== undefined)
```

Comment:
> Thanks for the contribution. I think it's better to set parser.decorators in [normalizeConfig](https://github.com/farm-fe/farm/blob/main/packages/core/src/config/index.ts#L246)

Thank you for you review. :) 

I'll refactor this now

Reply:
Hi, I was refactoring my code in the [`normalizeConfig`](https://github.com/farm-fe/farm/blob/main/packages/core/src/config/index.ts#L246) function and I realized that Typescript doesn't allow me to set one of the values in case of default non-optional parameters. Should I set a default value for them?

```ts
export interface ScriptParseConfig {
  esConfig?: {
    jsx?: boolean;
    fnBind: boolean;
    // Enable decorators.
    decorators?: boolean;

    // babel: `decorators.decoratorsBeforeExport`
    // Effective only if `decorator` is true.
    decoratorsBeforeExport: boolean;
    exportDefaultFrom: boolean;
    // Stage 3.
    importAssertions: boolean;
    privateInObject: boolean;
    allowSuperOutsideMethod: boolean;
    allowReturnOutsideFunction: boolean;
  };
  tsConfig?: {
    tsx: boolean;
    decorators?: boolean;
    /// `.d.ts`
    dts: boolean;
    noEarlyErrors: boolean;
  };
}
```

Reply:
I think to change these options as optional options